MAINT: remove deprecated in numpy/lib/_function_base_impl.py

Abhi210 opened the pull request:
This PR is related to gh-11521 and removes long-deprecated code paths in`numpy/lib/_function_base_impl`:

- Removed `numpy.trapz`, deprecated in NumPy 2.0 (2023-08-18)
- Removed `disp` method
- Removed `bias` and `ddof` arguments from `numpy.corrcoef`.
-  Removed associated tests from `numpy/lib/tests/test_function_base.py` and
  `numpy/_core/test_deprecations.py`

Associated tests and type stubs were also updated. All tests and Ruff checks pass locally.

<!--         ----------------------------------------------------------------
                MAKE SURE YOUR PR GETS THE ATTENTION IT DESERVES!
                ----------------------------------------------------------------

*  FORMAT IT RIGHT:
      https://www.numpy.org/devdocs/dev/development_workflow.html#writing-the-commit-message

*  IF IT'S A NEW FEATURE OR API CHANGE, TEST THE WATERS:
      https://www.numpy.org/devdocs/dev/development_workflow.html#get-the-mailing-list-s-opinion

*  HIT ALL THE GUIDELINES:
      https://numpy.org/devdocs/dev/index.html#guidelines

*  WHAT TO DO IF WE HAVEN'T GOTTEN BACK TO YOU:
      https://www.numpy.org/devdocs/dev/development_workflow.html#getting-your-pr-reviewed
-->

Abhi210 committed:
MAINT: remove deprecated `trapz` function, `disp` method and `bias/ddof` args in `corrcoef`

This commit removes long-deprecated code paths in `numpy/lib/_function_base_impl`:
- Removed `numpy.trapz`, deprecated in NumPy 2.0 (2023-08-18)
- Removed `disp` method
- Removed `bias` and `ddof` arguments from `numpy.corrcoef`
Removed associated tests from `numpy/lib/tests/test_function_base.py` and
  `numpy/_core/test_deprecations.py`

Associated tests and type stubs were also updated.
All tests and Ruff checks pass locally.

mattip commented:
LGTM. Please add a release note like [the one for 29986](https://github.com/numpy/numpy/blob/main/doc/release/upcoming_changes/29986.expired.rst?plain=1) but with this PR's number

jorenham reviewed (approved):
Looks good; thanks

Abhi210 committed:
DOC: add release note for removal of deprecated trapz, disp, and corrcoef args

jorenham commented on doc/release/upcoming_changes/29997.expired.rst:
```diff
@@ -0,0 +1,12 @@
+
+* ``numpy.trapz`` — deprecated since NumPy 2.0 (2023-08-18). Use ``numpy.trapezoid`` or
+  ``scipy.integrate`` functions instead.
+* ``disp`` method — deprecated in earlier releases and no longer functional. Use your own printing function instead.
```
I suppose `disp` is/was a more a function than it was a method

jorenham reviewed (commented):
I left some nitpickings

Abhi210 committed:
MAINT: remove redundant deprecated `bias` and `ddof` handling in `corrcoef`

Removes leftover deprecation logic for the `bias` and `ddof` arguments
in `numpy.corrcoef` from `extra.py`. These arguments were already
deprecated and removed from `_function_base_impl`.

No functional changes beyond removing redundant code. All tests and
Ruff checks pass locally.

Abhi210 committed:
MAINT: remove redundant deprecated `bias` and `ddof` handling in `corrcoef`

Removes leftover deprecation logic for the `bias` and `ddof` arguments
in `numpy.corrcoef` from `extra.py`. These arguments were already
deprecated and removed from `_function_base_impl`.

No functional changes beyond removing redundant code. All tests and
Ruff checks pass locally.

Abhi210 commented on doc/release/upcoming_changes/29997.expired.rst:
```diff
@@ -0,0 +1,12 @@
+
+* ``numpy.trapz`` — deprecated since NumPy 2.0 (2023-08-18). Use ``numpy.trapezoid`` or
+  ``scipy.integrate`` functions instead.
+* ``disp`` method — deprecated in earlier releases and no longer functional. Use your own printing function instead.
```
The comment says deprecarted in numpy 2.0

Abhi210 committed:
This commit removes remaining deprecated `bias` and `ddof` arguments from
`numpy.ma.extras` and updates the related tests.

In addition:
- Removed redundant declarations of `bias` and `ddof` in `extras.py`
  (deprecated since NumPy 2.0)
- Removed associated tests from `test_extras.py` and `test_regression.py`
- Amended `numpy/lib/_function_base_impl.py` to fully remove these arguments
  from function signatures and ensure consistency with documentation
- Documented the above changes in the release notes

All tests and Ruff checks pass locally.

jorenham commented on doc/release/upcoming_changes/29997.expired.rst:
```diff
@@ -0,0 +1,11 @@
+* ``disp`` function — deprecated from 2.0 release and no longer functional. Use your own printing function instead.
+* ``bias`` and ``ddof`` arguments in ``numpy.corrcoef`` — these had no effect since NumPy 1.10.
+
+
```
```suggestion
```

Abhi210 commented:
Hello!

I found that there were still some redundant declarations of the `bias` and `ddof` arguments in the `corrcoef` function within `extras.py`, so I’ve removed them. The corresponding tests were also removed from `test_extras.py`. Additionally, I had previously forgotten to remove `bias` and `ddof` from the `corrcoef` function signature itself. This has now been corrected.

Lastly, I’ve updated the release notes as suggested by @jorenham. Thank you for bringing these issues to my attention, and I apologize for the earlier oversight.

Abhi210 committed:
Update doc/release/upcoming_changes/29997.expired.rst

Co-authored-by: Joren Hammudoglu <jhammudoglu@gmail.com>

mattip committed:
doc: correct underline formatting

mattip commented on doc/release/upcoming_changes/29997.expired.rst:
```diff
@@ -0,0 +1,9 @@
+Removal of deprecated `trapz` function, `disp` function, and `bias`/`ddof` args in `corrcoef`
+--------------------------------------------------------------------------------------------
```
Simplify the heading
```suggestion
Removal of deprecated functions and arguments
---------------------------------------------
```

mattip committed:
Doc: simplify heading [skip actions][skip azp][skip cirrus]

mattip commented on doc/release/upcoming_changes/29997.expired.rst:
```diff
@@ -0,0 +1,12 @@
+Removal of deprecated `trapz` function, `disp` method, and `bias`/`ddof` args in `corrcoef`
```
I simplified the heading, it got too long.
